### ISSUE TITLE
Failure with jetbrains @Nullable annotation

### ISSUE THREAD

**Issue by FlorianCousin** [vojtechhabarta/typescript-generator#795]
I have a project in which I try to generate typescript DTO from Java one with some nullable fields.

I use jetbrains Nullable annotation which is annotated `@Target({ElementType.METHOD, ElementType.FIELD, ElementType.PARAMETER, ElementType.LOCAL_VARIABLE, ElementType.TYPE_USE})`.

However when I try to process classes with maven, I get the following error :
```
[ERROR] Failed to execute goal cz.habarta.typescript-generator:typescript-generator-maven-plugin:2.35.1025:generate (generate) on project TestTypescriptGenerator: Execution generate of goal cz.habarta.typescript-generator:typescript-generator-maven-plugin:2.35.1025:generate failed: 'org.jetbrains.annotations.Nullable' annotation cannot be used as nullable annotation because it doesn't have 'TYPE_PARAMETER' or 'TYPE_USE' target.
```

Moreover, I tried to create my own nullable annotation that I called `MyNullable` with the same target as the jetbrains nullable one, and the maven processing classes works without any error.

I have a sample project to reproduce the issue :
https://github.com/FlorianCousin/test-typescript-generator

Is there something I misunderstood in the available nullable annotations ?

**Comment by shmert** [comment]
Running into this as well. I suspect it's because the JetBrains annotation `Retention` is `CLASS` instead of `RUNTIME`. If so the error message is a little misleading.

```java
package org.jetbrains.annotations;

@Retention(RetentionPolicy.CLASS)
@Target({ElementType.METHOD, ElementType.FIELD, ElementType.PARAMETER, ElementType.LOCAL_VARIABLE, ElementType.TYPE_USE})
public @interface Nullable {
  // ...
}
```

I created my own annotation which is working well, using `RetentionPolicy.RUNTIME`:

```java
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.FIELD, ElementType.PARAMETER, ElementType.TYPE_PARAMETER, ElementType.TYPE_USE})
public @interface NullableType {
}
```

**Comment by FlorianCousin** [comment]
Thank you for your answer @shmert.

As we can see in my sample project, when I create my own nullable annotation with the same ones as the jetbrains annotation, that is the following annotations:
```java
@Documented
@Retention(RetentionPolicy.CLASS)
@Target({
  ElementType.METHOD,
  ElementType.FIELD,
  ElementType.PARAMETER,
  ElementType.LOCAL_VARIABLE,
  ElementType.TYPE_USE
})
```
Then the compilation works.

Therefore, I don't think the class retention is the same problem as the one I mentioned.

**Comment by vojtechhabarta** [comment]
There are two problems with JetBrains `@Nullable` annotation:
- old version without `TYPE_USE` is used because this project uses `kotlin-stdlib` which depends on very old `org.jetbrains:annotations`
- but more importantly: it doesn't have runtime retention

So I added warning that such annotation doesn't have any effect (and I also updated that dependency).

**Comment by vojtechhabarta** [comment]
Released in [3.1.1185](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v3.1.1185).
